release/nightly_tests/dataset/benchmark.py

```diff
@@ -85,6 +128,10 @@ def run_fn(self, name: str, fn: Callable[..., Any], **fn_run_args):
         print(f"Result of case {name}: {curr_case_metrics}")
 
     def write_result(self, output_path="/tmp/result.json"):
+        """Write all collected benchmark results to `output_path`.
+        The result is a dict of the form:
+        ``{case_name: {metric_name: metric_value, ...}}``."""
+
         test_output_json = os.environ.get("TEST_OUTPUT_JSON", output_path)
         with open(test_output_json, "w") as f:
             self.result["name"] = self.name
```

Comment:
Nit: Would move these lines to a separate function. Right now, reporting completion of a benchmark is coupled with writing the results. 

```python
        print(
            f"Finished benchmark {self.name}, metrics exported to {test_output_json}:"
        )
        print(self.result)
```

We could then add a `benchmark.finalize` or something like that:

```
def finalize():
    write_results(...)
    report_results()
```

Reply:
My intention here was to ensure we also print the metrics to the log file whenever we write metrics to an external file. I found that there is sometimes an issue where the test succeeds, but metrics do not get properly populated to our internal release test results database, so by including the print logic inside the `write_results()` function, we guarantee that we'll at least have some backup output for every `write_results()` call.